sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -418,27 +419,53 @@ case class AlterTableRenamePartitionCommand(
  */
 case class AlterTableDropPartitionCommand(
     tableName: TableIdentifier,
-    specs: Seq[TablePartitionSpec],
+    specs: Seq[Expression],
     ifExists: Boolean,
     purge: Boolean)
-  extends RunnableCommand {
+  extends RunnableCommand with PredicateHelper {
+
+  private def hasComplexExpr(expr: Expression): Boolean = {
```

Comment:
The function name looks confusing. Actually they are not more complex operators, are they?